### DISCUSSION TITLE
Feature Request: Read an indexed JSON file

### DISCUSSION THREAD

**Discussion by janrito** [duckdb/duckdb#9834] [Ideas]
JSON files come in all shapes. One that I've recently had to work with is where a single json object represents the index of a table, and each value in that object represents a record. i.e.:

```json
{
  "pk0001": {"name": "Bob", "email": "bob@corp.com"}
  "pk0002": {"name": "Alice", "email": "alice@corp.com"}
  "pk0003": {"name": "Charlie", "email": "charlie@corp.com"}
  ...
}
```

In it's simplest form, this can be helpful in encoding datasets where there is a natural unique constraint — a key/value dataset.

Reading this currently is possible using unpivot:

```sql
WITH key_val AS (
    UNPIVOT
        read_json(
            "s3://bucket/path/to/dataset.json.gz",
            auto_detect=True,
            sample_size=-1,
            format=newline_delimited
        )
    ON COLUMNS(*)
    INTO
        NAME key
        VALUE val
)

SELECT key, val FROM key_val
```

It's pretty cumbersome – but also, it will run out of memory on larger datasets.

As an added argument, Pandas lists `index` as one of the standard ways in which json is oriented. 

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_json.html
> orient str
> ...
> The format of the JSON string:
> ...
>‘index’ : dict like {index -> {column -> value}}

**Comment by thohan88** [comment]
Agree! On the same note, the ability to store this type of data in an index/value-table in an efficient way. I recently tried this on a dataset with 70K records where each index was an integer and the value was a ~1MB JSON-object. Slow performance and minimal achieved compression compared to e.g. Snowflake XS.
